### PR TITLE
Add 32f_s32f_add_32f kernel

### PR DESCRIPTION
**Pull request by ghostop14**
This kernel adds in vector + scalar functionality

### CODE REVIEW

**Comment by ghostop14** [comment]
If you wouldn't mind merging this one first, I'll incorporate it into #364 before merging that one.  Thanks!

**Comment by argilo** [comment]
CI fails with this error:
 ```
libvolk.so.2.2.1: undefined reference to `volk_32f_s32f_add_32f_a_orc_impl'
```

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (argilo) [inline kernels/volk/volk_32f_s32f_add_32f.h]: This example is no longer normalizing / scaling the input, so this example should be updated.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (argilo) [inline kernels/volk/volk_32f_s32f_add_32f.h]:
```suggestion
    cVal = vaddq_n_f32 (aVal, scalar); // Do the add
```

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (argilo) [inline kernels/volk/volk_32f_s32f_add_32f.h]: This typo was copied over from the scalar multiply kernel:
```suggestion
#endif /* LV_HAVE_ORC */
```

**Comment by argilo** [comment]
The reason for the CI failure is that `kernels/volk/asm/orc/volk_32f_s32f_multiply_32f_a_orc_impl.orc` was not copied over and adapted for addition. Either that should be done, or the `#ifdef LV_HAVE_ORC` section should be removed.

**Comment by argilo** [comment]
I notice there are two separate include guards: `INCLUDED_volk_32f_s32f_add_32f_u_H` and `INCLUDED_volk_32f_s32f_add_32f_a_H`. The NEON kernel is in the latter section despite being an unaligned kernel. (This happened because it's the same in the multiply const kernel.)

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (michaelld) [inline kernels/volk/volk_32f_s32f_add_32f.h]: just the current year for a new file

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (michaelld) [inline kernels/volk/volk_32f_s32f_add_32f.h]: remove the extra line here

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (michaelld) [inline kernels/volk/volk_32f_s32f_add_32f.h]: is there an aligned version for neon?

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (michaelld) [inline kernels/volk/volk_32f_s32f_add_32f.h]: extra space between "f32" and "(" .. not a big deal since we'll do a clang-format soon-ish ... but still like to get it fixed where reasonable

**Review by michaelld — CHANGES_REQUESTED** [review]
A few minor changes. looks pretty good right already though!

**Comment by ghostop14** [comment]
@michaelld all cleaned up.  On the a_neon question, this set was basically a copy-and-change version of the existing multiply block to get the add functionality.  Multiply didn't have one to use as a foundation either.

**Comment by michaelld** [comment]
> @michaelld all cleaned up. On the a_neon question, this set was basically a copy-and-change version of the existing multiply block to get the add functionality. Multiply didn't have one to use as a foundation either.

OK; that makes sense about the neon. 'u' is a superset of 'a' anyway, so having 'u' working is the way to go if there is just 1 implementation.

**Review by michaelld — APPROVED** [review]
LGTM

**Comment by michaelld** [comment]
@jdemel Let's get your review here to get this PR finalized, so that @ghostop14 can then integrate it into #364 & we can move on that.

**Comment by ghostop14** [comment]
Argilo and I actually removed the requirement in 364.  The add 1e-20 wasn't necessary.  So they can actually be approved separately (no longer coupled).

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: I suggest, we remove the "aligned" version.

As far as I recall, we don't use separate include guards for aligned and unaligned kernels. Every kernel just gets its `#ifdef LV_HAVE_*`. That's probably an artifact from former times.

A few more things regarding the generic kernel.
1. Please move the generic kernel to the top. In most cases it serves as further documentation for others who want to understand the kernel in depth.
2. Let's just have one `generic` kernel and not aligned and unaligned.
3. Please use the generic kernel to take care of all the remainders in the other kernels. Mostly, because this avoids copypasta. I know that's not much in this case but I want to stick to good code practice and all code should serve as an example. Especially whenever someone's working on it.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: Just use the generic kernel here.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: Just use the generic kernel here

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: I suggest to remove the extra include guards for aligned/unaligned kernels.
In this case, we can avoid these double includes.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: Please use the generic kernel here.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: This generic kernel should be sufficient. No need for an aligned generic version.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: Please use the generic kernel.

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: Please use the generic kernel

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h]: Please consolidate these include guards such that we only have one for all kernel versions.

**Review by jdemel — CHANGES_REQUESTED** [review]
Overall this PR looks good to me. Just a few minor things. Thanks for your patience and your work.

**Comment by ghostop14** [comment]
@jdemel, just pushed updates.  I think I checked off all your comments.  Let me know if I missed anything?

**Comment by ghostop14** [comment]
Rebased, fixed merge issues, clang-formatted my modified files, and re-pushed.  Doesn't look like CI re-ran after force push but the conflicts are resolved in the repo.

**Comment by jdemel** [comment]
There seems to be an issue with GitHub. These are the current stats: `+32,918 −27,686`. I'd be really hesitant to just hit the merge button here. I assume git will be more helpful here.

**Comment by michaelld** [comment]
Looks like the rebase didn't fully succeed ... please give it a go again when you get a chance!

File: kernels/volk/volk_32f_s32f_add_32f.h
Comment (jdemel) [inline kernels/volk/volk_32f_s32f_add_32f.h:192]: The include guards should be consolidated into one. For me, this is mainly a matter of consistency and it would be great to have it in here.

**Review by jdemel — CHANGES_REQUESTED** [review]
I have one, hopefully minor comment and then this PR is good to go.

**Comment by ghostop14** [comment]
@jdemel take a look at this version.  I also rebased on master before pushing for this one.  I think the guards were okay for aligned versus unaligned with the exception of the ORC kernels.  Both aligned and unaligned ORC functions were in the aligned section, so I fixed that.  Not sure if that was exactly what you saw or not.  FYI, this kernel was templated off of the scalar multiply kernel, so the same guard issue is probably in some of the other scalar kernels too.  (Future cleanup task :) )

**Comment by jdemel** [comment]
@ghostop14 Thanks for the rebase. Besides, I think there's a reason for ORC kernels to have aligned/unaligned include guards. At least, CI tells us that the kernel is broken now. It won't even compile on ARM. I wonder why though. Anyways. I assume, this is something we need to investigate at a different time. 
I'm OK with this PR. I guess you can revert your last commit where you moved the ORC kernel function.

**Comment by ghostop14** [comment]
@jdemel done.  We'll just make sure it builds in CI (famous last words... built fine on my machine)

**Comment by michaelld** [comment]
CI error that I think needs to be fixed up herein:
```
In file included from /home/travis/build/gnuradio/volk/build/lib/volk_machine_neon.c:81:0:
/home/travis/build/gnuradio/volk/kernels/volk/volk_32f_s32f_add_32f.h: In function ‘volk_32f_s32f_add_32f_u_neon’:
/home/travis/build/gnuradio/volk/kernels/volk/volk_32f_s32f_add_32f.h:184:35: error: ‘cPtr’ undeclared (first use in this function)
     volk_32f_s32f_add_32f_generic(cPtr, aPtr, scalar, num_points - number);
                                   ^
/home/travis/build/gnuradio/volk/kernels/volk/volk_32f_s32f_add_32f.h:184:35: note: each undeclared identifier is reported only once for each function it appears in
/home/travis/build/gnuradio/volk/kernels/volk/volk_32f_s32f_add_32f.h:184:41: error: ‘aPtr’ undeclared (first use in this function)
     volk_32f_s32f_add_32f_generic(cPtr, aPtr, scalar, num_points - number);
                                         ^
make[2]: *** [lib/CMakeFiles/volk_obj.dir/volk_machine_neon.c.o] Error 1
make[1]: *** [lib/CMakeFiles/volk_obj.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by ghostop14** [comment]
@michaelld looks good now.

**Review by jdemel — APPROVED** [review]
LGTM. Thanks for working on it!

**Review by michaelld — APPROVED** [review]
LGTM

**Comment by michaelld** [comment]
Relevant CI passes. 2 approvals. Nice addition! Merging

**Pull request merged**
